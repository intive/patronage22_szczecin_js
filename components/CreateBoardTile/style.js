import styled from 'styled-components'

export const Tile = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16rem;
  height: 12.5rem;
  background-color: #E1E1E1;
  border-radius: 8px;
  cursor: pointer;
`
