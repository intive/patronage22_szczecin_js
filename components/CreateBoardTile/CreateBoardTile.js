import Button from '../Button/Button'
import { Tile } from './style'
import BoardCreator from '../BoardCreator/BoardCreator'
import useModal from '../../hooks/useModal'

export default function CreateBoardTile () {
  const { isOpen, toggle } = useModal()

  return (
    <>
      <Tile role='button' tabIndex='0' onClick={toggle} data-testid='create-board-tile'>
        <Button icon='crop_original'>New board</Button>
      </Tile>
      <BoardCreator onClose={toggle} isOpen={isOpen} />
    </>
  )
}
