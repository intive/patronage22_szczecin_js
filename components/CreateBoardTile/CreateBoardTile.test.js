/* globals describe, it, expect  */
/* eslint-env jest */

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import CreateBoardTile from './CreateBoardTile'

describe('CreateBoardTile', () => {
  it('renders correctly', () => {
    const { container } = render(<CreateBoardTile />)

    expect(container).toMatchSnapshot()
  })

  it('should open modal when button is clicked', () => {
    render(<CreateBoardTile />)
    userEvent.click(screen.getAllByRole('button')[0])

    expect(screen.getByRole('button', { name: /buttons.continue/ })).toBeInTheDocument()
  })

  it('should open modal when whole tile is clicked', () => {
    render(<CreateBoardTile />)
    userEvent.click(screen.getByText('New board'))

    expect(screen.getByRole('button', { name: /buttons.continue/ })).toBeInTheDocument()
  })

  it('should close open modal when close button is clicked', () => {
    render(<CreateBoardTile />)
    userEvent.click(screen.getByText('New board'))
    const button = screen.getByRole('button', { name: /close/ })
    userEvent.click(screen.getByRole('button', { name: /close/ }))

    expect(button).not.toBeInTheDocument()
  })
})
