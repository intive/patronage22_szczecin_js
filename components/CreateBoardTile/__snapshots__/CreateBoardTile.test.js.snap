// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBoardTile renders correctly 1`] = `
<div>
  <div
    class="sc-dkPtRN hzCcxk"
    role="button"
    tabindex="0"
  >
    <button
      class="sc-bdvvtL dRBcfm"
    >
      <span
        class="material-icons sc-gsDKAQ jUyrhl"
      >
        crop_original
      </span>
      New board
    </button>
  </div>
</div>
`;
